### PR TITLE
[risk=no] [RW-6472] React age chart component (fitbit-view)

### PR DESCRIPTION
**Pull request by SrushtiGangireddy**
Converting the age chart component in fitbit view.
Also fixing update analysis issue on highcharts component.

### CODE REVIEW

File: public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx [hunk 1]
```diff
@@ -24,13 +26,21 @@ export class BioSexChartReactComponent extends React.Component<Props, State> {
 
   constructor(props) {
     super(props);
-    this.state = {options: null};
+    this.state = {options: null, genderAnalysis: props.genderAnalysis,
+    genderCountAnalysis: props.genderCountAnalysis};
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx]: Same as comment above, avoid copying props into state if possible

File: public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx [hunk 2]
```diff
@@ -24,13 +26,21 @@ export class BioSexChartReactComponent extends React.Component<Props, State> {
...
 
   componentDidMount() {
       this.getChartOptions();
   }
 
+  componentDidUpdate(prevProps: Readonly<Props>) {
+    const newAnalysis = this.props.genderAnalysis;
+    if (newAnalysis !== this.state.genderAnalysis) {
+        this.setState({ genderAnalysis: newAnalysis }, this.getChartOptions);
+    }
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx]: Same as suggestion above:
```suggestion
    if (prevProps.genderAnalysis !== this.props.genderAnalysis) {
        this.getChartOptions();
    }
```

File: public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        pointData.sort((a, b) => a.name.localeCompare(b.name));
+        categoryArr.sort();
+        return { categories: categoryArr, data: pointData};
+  }
+
+  render() {
+    const {options} = this.state;
+    return <div>
+          {options && <HighchartsReact highcharts={highCharts} options={options}
+          updateArgs={[true]}/>}
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx:103]: Nice find adding `updateArgs`. I hadn't seen that before but seems to work well for updating the chart when props change.

File: public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+  componentDidMount() {
+    this.getChartOptions();
+  }
+
+  componentDidUpdate(prevProps: Readonly<Props>) {
+    const newAnalysis = this.props.ageAnalysis;
+    if (newAnalysis !== this.state.ageAnalysis) {
+        this.setState({ ageAnalysis: newAnalysis }, this.getChartOptions);
+    }
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx]: We can compare current props with `prevProps` arg to check for changes:
```suggestion
    if (prevProps.ageAnalysis !== this.props.ageAnalysis) {
        this.getChartOptions();
    }
```

File: public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    selectedResult: any;
+    domain: string;
+}
+
+export class AgeChartReactComponent extends React.Component<Props, State> {
+
+  constructor(props) {
+    super(props);
+    this.state = {options: null, ageAnalysis: props.ageAnalysis,
+    ageCountAnalysis: props.ageCountAnalysis};
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx]: Copying props into state is not recommended (see the note in the [React docs for constructor](https://reactjs.org/docs/react-component.html#constructor)). We should be able to do what we need without it anyway, see the comment below in `componentDidUpdate`

File: public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx (hunk 1 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx]: Thanks, changed this.

File: public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx (hunk 2 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx]: Thanks, changed this.

File: public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx (hunk 5 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx]: Thanks, changed this.

**Review by dolbeew — APPROVED** [review]

**Pull request merged**